Clean up merge modal and allow "live-merge" of explorative and project

philippotto opened the pull request:
### URL of deployed dev instance (used for testing):
- https://cleanupmergemodal.webknossos.xyz

### Steps to test:
- merge an explorative directly in the tracing view (new functionality) via merge-modal
- merge an explorative directly old-school (creates a new explorative)
- do the same for a project

### To discuss:
- for big projects (or exploratives), the merge will probably work, but saving to the server might crash. we probably need to think about how to tackle this (e.g., chunk the update actions so the server is not overwhelmed; show better progress report so that saving for a few minutes is an ok UX; show a warning that this may take some time etc...)

### Issues:
- fixes #2930 
- fixes #2929 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [X] ~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~
- [X] Ready for review

daniel-wer commented on app/assets/javascripts/oxalis/view/action-bar/merge_modal_view.js:
```diff
@@ -101,98 +108,91 @@ class MergeModalView extends PureComponent<Props, MergeModalViewState> {
+      this.setState({ isUploading: false });
+      Toast.success(messages["tracing.merged"]);
+    } else {
+      Toast.error("Merging is not supported for volume tracings.");
```
Should this be moved to the messages file as well? Actually, I'm not sure what our current practice regarding the messages.js is anyways? Maybe we should discuss this, not as part of this PR :)

daniel-wer commented:
Also, does the backend still do some kind of tree renaming when merging a project @fm3? If so, does this happen in this "frontend-merge" case as well (by retrieving the compound project from the server)? I just want to make sure that we don't lose any functionality :)

fm3 commented:
no, there is no tree renaming, only the node id mapping and the combining of the tracings’ bounding boxes

daniel-wer commented:
Sorry for spamming, but I just noticed another thing. When un-checking the checkbox (to create a new tracing when merging), didn't we redirect to that new tracing before? I didn't use the merge functionality for quite a while now, so maybe that wasn't the case, but it feels weird that the user just gets the "Merging successful" message and then nothing happens (apart from the new tracing appearing in the explorative tracings list, which the user cannot see at that point).

philippotto commented:
Thank you for your feedback, @daniel-wer.

> I would expect the merge modal to close after a successful merge.

I guess this has never been like that for the case that the user wants to do several merges consecutively. I wouldn't change it now unless some requests it.

> It's also possible to merge projects/explorative annotations from other datasets. I'm not sure whether this was allowed before (probably?), but I don't think this is wanted behavior and we specifically disallow this when uploading NMLs.

Good point, I added a guarding check for that.

> When un-checking the checkbox (to create a new tracing when merging), didn't we redirect to that new tracing before?

Good point! In fact, the code redirects to the new tracing via the router's history.push function, but this doesn't work (I think, since the oxalis controller does not listen to prop changes..). I changed the redirect to use `location.href` so that we get a fresh page.

philippotto commented:
@daniel-wer @fm3 What do you think about the bounding boxes? When doing the merge via the back-end, the bboxes are combined. When done locally, it's only importing the trees. I'm not sure whether this is an issue at all. The trees will be visible, anyway..

Also, do we want to add a "warning" that the local change can take some time for saving. Or do we want to land this feature together with the chunk saving PR which adds progress indicators.

daniel-wer commented:
I'd say it's fine to ignore the bounding boxes when merging in the frontend for now. As you say, the trees are visible anyways. We can create an issue and add it later.
Let's land it together with the progress indicator as that PR is also almost ready.
